### DISCUSSION TITLE
ILI9481 8 bit parallel

### DISCUSSION THREAD

**Discussion by NotABadger** [moononournation/Arduino_GFX#264] [Q&A] [answered]
Hello fellow engineers,

I'm currently working on a project in which I would like to use a screen (see picture) 
![s-l1600](https://user-images.githubusercontent.com/10919787/222205612-78610e4f-4265-4fd9-887f-06bcee2d7cd2.jpg)

I would like to use the 8-bit parallel interface since I need some pins for I²C & battery monitoring. So i'm using "Arduino_DataBus *bus = new Arduino_ESP32PAR8(27 /* DC */, 5 /* CS */, 26 /* WR */, GFX_NOT_DEFINED /* RD */, 16 /* D0 */, 17 /* D1 */, 18 /* D2 */, 19 /* D3 */, 21 /* D4 */, 12 /* D5 */, 23 /* D6 */, 25 /* D7 */);" [as shown on the wiki](https://github.com/moononournation/Arduino_GFX/wiki/Data-Bus-Class#8-bit) for the bus.

I found the Arduino_ILI9481_18_bit in the documentation, which  according to the wiki: "This display SPI interface not support 16-bit SPI. It require convert from 16-bit to 18-bit for every pixels so it is not very efficient." however, it also doesn't work with the 8-bit parallel bus.

there is no Arduino_ILI9481 (regular) in the code, only in the documentation: https://github.com/moononournation/Arduino_GFX/wiki/Display-Class#ili9481-lcd-parellel-interface-320x480
Is there a reason that it is no longer available in the library? and is there a work around to make the ILI9481 work with an 8-bit parallel interface? I would really want to make it work with the parallel interface because (i've tried) SPI has too low of a refresh rate for my intentions.
 
Can anyone help me?

**Comment by moononournation** [comment, marked as answer]
Just because I did not have a ILI9481 using 8-bit display board in hand. Your display board is for Mega2560 and have 5V voltage converter. I don't know is it fit for a 3.3V MCU like ESP32.

**Reply by NotABadger** [reply]
Oh, yes ofcourse. You're right. 
My first screen for testing the SPI interface was based on an ILI9488, and according to the data sheet that one only used 5V for backlight.
This screen did not have a data sheet, but I should have realized that the logic pins would also require 5V for this screen.
There are some logic MOSFET's comming in real soon, and I'll try it again when those are in.

Thanks for your help!
Now i'm off to see if I can find a block of lead in the attic to keep me company, because apparently we're both pretty dense.

**Reply by NotABadger** [reply]
I put the display under my microscope, and it is using the [74HC245 ](https://assets.nexperia.com/documents/data-sheet/74HC_HCT245.pdf) transceiver to interface with the ILI9481, Which is indeed a 5V switching IC. It needs ~3.5V to switch to a high state when using 5V (see page 5/15 of the linked data sheet).
